heat/core/logical.py

```diff
@@ -516,10 +517,11 @@ def sanitize_input_type(
     x = sanitize_input_type(x, y)
     y = sanitize_input_type(y, x)
 
-    # if one of the tensors is distributed, unsplit/gather it
-    if x.split is not None and y.split is None:
-        t1 = manipulations.resplit(x, axis=None)
-        return t1, y
+    # if one of the DNDarrays is distributed and the other is not
+    if x.split is not None and y.split is None and y.ndim > 0:
```

Comment:
```suggestion
    if x.is_distributed() and not y.is_distributed() and y.ndim > 0:
```
if `split is not None` but users are running things on 1 MPI process, the array is not distributed and no need for the extra steps